drivers/usb_c/tcpc/ucpd_stm32.c

```diff
@@ -1023,15 +1023,24 @@ static void ucpd_isr(const struct device *dev_inst[])
 	 */
 
 	uint32_t ucpd_base;
+	uint32_t sr1;
+	uint32_t sr2;
 
 	/*
 	 * Since the UCPD peripherals share the same interrupt line, determine
 	 * which one generated the interrupt.
 	 */
-	if (LL_SYSCFG_IsActiveFlag_UCPD1()) {
+
+	/* Read UCPD1 Status Register */
+	sr1 = LL_UCPD_ReadReg(((const struct tcpc_config *)dev_inst[0]->config)->ucpd_port, SR);
+
+	/* Read UCPD2 Status Register */
+	sr2 = LL_UCPD_ReadReg(((const struct tcpc_config *)dev_inst[1]->config)->ucpd_port, SR);
+
+	if (sr1) {
 		/* UCPD1 interrupt is pending */
 		ucpd_base = UCPD1_BASE;
```

Comment:
As a follow up PR, you can eliminate the device instance lookup at line 1055 by setting the `dev` and `config` locals directly in this branch.

```
if (sr1) {
    dev = dev_inst[0];
} else if (sr2) {
    dev = dev_inst[1];
} else {
    return;
}
config = dev->config

Reply:
Done